scipy/stats/tests/test_multivariate.py

```diff
@@ -2651,72 +2636,142 @@ def method(x):
         assert_equal(rvs.sum(axis=-2), np.broadcast_to(col, (1000, 3)))
 
-    def test_rvs_size(self):
+    @pytest.mark.parametrize("method", ("boyett", "patefield"))
+    def test_rvs_size(self, method):
         row = [2, 6]
         col = [1, 3, 4]
 
         # test size `None`
-        rng = np.random.default_rng(628174795866951638)
-        rv = random_table.rvs(row, col, random_state=rng)
+        rv = random_table.rvs(row, col, method=method,
+                              random_state=self.get_rng())
         assert rv.shape == (2, 3)
 
         # test size 1
-        rng = np.random.default_rng(628174795866951638)
-        rv2 = random_table.rvs(row, col, size=1, random_state=rng)
+        rv2 = random_table.rvs(row, col, size=1, method=method,
+                               random_state=self.get_rng())
         assert rv2.shape == (1, 2, 3)
         assert_equal(rv, rv2[0])
 
         # test size 0
-        rv3 = random_table.rvs(row, col, size=0, random_state=rng)
+        rv3 = random_table.rvs(row, col, size=0, method=method,
+                               random_state=self.get_rng())
         assert rv3.shape == (0, 2, 3)
 
         # test other valid size
-        rng = np.random.default_rng(628174795866951638)
-        rv4 = random_table.rvs(row, col, size=20, random_state=rng)
+        rv4 = random_table.rvs(row, col, size=20, method=method,
+                               random_state=self.get_rng())
         assert rv4.shape == (20, 2, 3)
 
-        rng = np.random.default_rng(628174795866951638)
-        rv5 = random_table.rvs(row, col, size=(4, 5), random_state=rng)
+        rv5 = random_table.rvs(row, col, size=(4, 5), method=method,
+                               random_state=self.get_rng())
         assert rv5.shape == (4, 5, 2, 3)
 
         assert_allclose(rv5.reshape(20, 2, 3), rv4, rtol=1e-15)
 
         # test invalid size
         message = "`size` must be a non-negative integer or `None`"
         with pytest.raises(ValueError, match=message):
-            random_table.rvs(row, col, size=-1, random_state=rng)
+            random_table.rvs(row, col, size=-1, method=method,
+                             random_state=self.get_rng())
 
         with pytest.raises(ValueError, match=message):
-            random_table.rvs(row, col, size=np.nan, random_state=rng)
+            random_table.rvs(row, col, size=np.nan, method=method,
+                             random_state=self.get_rng())
 
-    def test_frozen(self):
-        rng1 = np.random.default_rng(628174795866951638)
-        rng2 = np.random.default_rng(628174795866951638)
+    @pytest.mark.parametrize("method", ("boyett", "boyett2", "patefield"))
+    def test_rvs_method(self, method):
+        row = [1, 3]
+        col = [2, 1, 1]
+
+        ct = random_table
+        rvs = ct.rvs(row, col, size=10000, method=method,
+                     random_state=self.get_rng())
+
+        unique_rvs, counts = np.unique(rvs, axis=0, return_counts=True)
```

Comment:
Yes. It is a bit of a circular argument, we check pmf with the help of random samples and vice versa. If you have a better idea, let me know. I think it is ok, because it is very unlikely that we make two mistakes which somehow compensate so that pmf and random sampling agree although both are wrong.

Reply:
I think this is fine, but I think we did something very similar in `test_pmf_logpmf`. If we are doing check the relationship in the opposite direction somehow, that's fine, but maybe I missed it. As an example, if this were a univariate distribution, we might do check `cdf(ppf(p)) = p`, but we should also check `ppf(cdf(x)) = x`. To me, this looked like were applying the same approach twice, though.

Reply:
The test is very similar, but it is not redundant. I added some comments in the code to explain this. We cannot do the equivalent of `cdf(ppf(p)) = p` and `ppf(cdf(x)) = x` here, because random sample generation and pmf computation are not inverses of each other.